### PR TITLE
Circular Imports and Typing

### PR DESCRIPTION
**Pull request by PretzelVector**
Definitely a [seagull](https://changingminds.org/disciplines/leadership/articles/seagull_management.htm) PR.

I was unable to utilize this library due to #148 so I tried using HEAD, but had problems with circular imports.

Along the way, also added types I was interested in for what I was doing.

Works For Me™️

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -9,6 +9,7 @@ tidalapi
 
 Unofficial Python API for TIDAL music streaming service.
 
+Requires Python 3.7 or higher.
```
Comment (tehkillerbee) [inline README.rst:12]: This might change very soon.. We should remember to update it to the latest supported version.

**Comment by 2e0byo** [comment]
How did you trigger circular imports?  Would like to reproduce.

**Comment by PretzelVector** [comment]
> How did you trigger circular imports? Would like to reproduce.

```
$ git clone https://github.com/tamland/python-tidal.git
$ cd .\python-tidal\
$ git describe
v0.7.0-63-g53e7081
$ poetry install --no-root
$ python
Python 3.10.3 (tags/v3.10.3:a342a49, Mar 16 2022, 13:07:40) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import tidalapi
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\tmp\python-tidal\tidalapi\__init__.py", line 4, in <module>
    from .media import Track, Video  # noqa: F401
  File "C:\tmp\python-tidal\tidalapi\media.py", line 34, in <module>
    class Media:
  File "C:\tmp\python-tidal\tidalapi\media.py", line 63, in Media
    self, session: tidalapi.session.Session, media_id: Optional[str] = None
AttributeError: partially initialized module 'tidalapi' has no attribute 'session' (most likely due to a circular import)
>>>
```

**Comment by 2e0byo** [comment]
Oh rats that should never have happened, I really need to get the tests running again.

I'd prefer to avoid some of these long `x.y.z` but lots of refactoring will land soon anyhow.  Thanks for the improvements; merging.

**Pull request merged**